test/e2e/network/pod_lifecycle.go

```diff
@@ -0,0 +1,276 @@
+						Ports: []v1.ContainerPort{
+							{
+								ContainerPort: 80,
+							},
+						},
+						ReadinessProbe: &v1.Probe{
+							ProbeHandler: v1.ProbeHandler{
+								HTTPGet: &v1.HTTPGetAction{
+									Path: "/readyz",
+									Port: intstr.IntOrString{
+										IntVal: int32(80),
+									},
+									Scheme: v1.URISchemeHTTP,
+								},
+							},
+						},
+						LivenessProbe: &v1.Probe{
+							ProbeHandler: v1.ProbeHandler{
+								HTTPGet: &v1.HTTPGetAction{
+									Path: "/healthz",
+									Port: intstr.IntOrString{
+										IntVal: int32(80),
+									},
+									Scheme: v1.URISchemeHTTP,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		webserverPod.Spec.TerminationGracePeriodSeconds = utilpointer.Int64(gracePeriod)
+		webserverPod = podClient.CreateSync(ctx, webserverPod)
+
+		ginkgo.By("Creating 1 pause pods that will try to connect to the webserver")
+		pausePod := e2epod.NewAgnhostPod(ns, "pause-pod-1", nil, nil, nil)
+		pausePod = podClient.CreateSync(ctx, pausePod)
+
+		ginkgo.By("Try to connect to the webserver")
+		// Wait until we are able to connect to the Pod
+		podIPAddress := net.JoinHostPort(webserverPod.Status.PodIP, strconv.Itoa(80))
+		cmd := fmt.Sprintf(`curl -q -s --connect-timeout 5 %s/hostname`, podIPAddress)
+		if pollErr := wait.PollImmediate(framework.Poll, time.Minute, func() (bool, error) {
+			stdout, err := e2eoutput.RunHostCmd(pausePod.Namespace, pausePod.Name, cmd)
+			if err != nil {
+				framework.Logf("error connecting to pod address %s: %v", err, podIPAddress)
+				return false, nil
+			}
+			if webserverPod.Name != strings.TrimSpace(stdout) {
+				framework.Logf("pod expected hostname: %s  got: %v", webserverPod.Name, stdout)
+				return false, nil
+			}
+			return true, nil
+		}); pollErr != nil {
+			framework.ExpectNoError(pollErr, "pod not service traffic")
+		}
+		// webserver should continue to serve traffic after delete
+		// since will be gracefully terminating
+		err := cs.CoreV1().Pods(ns).Delete(ctx, webserverPod.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "error deleting webserver pod")
+		// wait some time to ensure the test hit the pod during the terminating phase
+		time.Sleep(15 * time.Second)
+		if pollErr := wait.PollImmediate(framework.Poll, time.Minute, func() (bool, error) {
+			stdout, err := e2eoutput.RunHostCmd(pausePod.Namespace, pausePod.Name, cmd)
+			if err != nil {
+				framework.Logf("error connecting to pod address %s: %v", err, podIPAddress)
+				return false, nil
+			}
+			if webserverPod.Name != strings.TrimSpace(stdout) {
+				framework.Logf("pod expected hostname: %s  got: %v", webserverPod.Name, stdout)
+				return false, nil
+			}
+			return true, nil
+		}); pollErr != nil {
+			framework.ExpectNoError(pollErr, "pod not service traffic")
+		}
+		// wait some time to ensure the test hit the pod during the terminating
+		time.Sleep(15 * time.Second)
```

Comment:
this feels gratuitous - after 15 seconds, it is surely terminating - do we need to do it again?

Reply:
paranoid mode for testing, will remove